README.md

```diff
@@ -20,20 +20,15 @@ You must have an Azure account and subscription. You must have a valid registrat
 
 **NOTE:** To learn how to deploy CAS on Microsoft Azure go to [Deploy Teradici Cloud Access Software on Azure.](https://github.com/teradici/pcoip-agent-azure-templates/blob/master/README.md)
 
-By default, the CAM deployment scripts will create a service principal account for CAM to use after deployment. In order for the CAM deployment scripts to create this service principal, you must pass an account to the **AzureAdminUsername** parameter which meets the following criteria:
+By default, the CAM deployment scripts will create a service principal account for CAM to use after deployment. In order for the CAM deployment scripts to create this service principal, you must sign into Azure with an account which meets the following criteria:
 1. The account must have owner access to the subscription (to be able to set access policies)
 1. The account must be able to be programatically logged in without user interaction. This means:
    1. It must be an organizational account, not a Microsoft account.
    1. It must not require multi-factor authentication.
 
 If you are not in posession of an account which meets the criteria, you must create a service principal account before deploying CAM.
 
-There are multiple ways to manually create a service principal account. See [Creating a Service Principal Account](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal) for instructions on how to do it from the Azure portal. Once the service principal account has been created, complete the following steps so that CAM can use the account to sign in:
-1. Give the service principal contributor access to a new resource group.
```

Comment:
I'm not sure how we could fix this in the powershell. Do you mean if the user provides their own SP creds, we'll have the deploy cam add in the appropriate permissions? If this is the case, I think we should add in the appropriate documentation and leave the script as is. I feel like we might get weird errors with insufficient privileges if the logged in cloud shell user doesn't have the required permissions to modify SP creds and stuff that we'd probably have to document and test.

Reply:
Yeah... The alternate is weird failures because they didn't give the SP the correct access rights. I think this is probably very simply doable. Like today - if Deploy-CAM creates the SP then it gives it contributor access rights to the target resource group. If not it doesn't do anything. I think we could check if the provided SP has contributor rights to the resource group at least and either fail with a good error message or (I think this is simple) try to add the role to the SP and then if that fails, fail with a good error message. I think if we do nothing we'll get an error message from the new-keyvault part today. (So errors will be logged :) )

Reply:
In that case should I leave this as is and add in the adding permissions stuff or leave it as future work? I can leave in the note about contributor access to the resource group.

Reply:
Yeah okay, let's assume no code changes and just document that if they are pre-creating the SP they need to give it contributor access to the resource group. They then need to pre-create the resource group as well, assign rights, and then tell CAM to deploy into that resource group by following the prompts.